File: util/template/TemplateModule/TemplateModule.cpp
/*
 * @date Created on [TODO-DATE].
 * @author [TODO-AUTHOR]
 */

#include "TemplateModule.hpp"


TemplateModule::TemplateModule(uint64_t instanceId) :
        AbstractModule(zmf::data::ModuleUniqueId(MODULE_TYPE_ID, instanceId), MODULE_VERSION, "TemplateModule",
                       std::vector<zmf::ModuleDependency>()) { }

TemplateModule::~TemplateModule() {
}


bool TemplateModule::enable() {
    // TODO Enable your module (eg. start threads, subscribe), return false if enabling fails
    return true;
}


void TemplateModule::disable() {
    // TODO Disable your module here (eg. stop threads)
}


void TemplateModule::handleModuleStateChange(std::shared_ptr<zmf::data::ModuleHandle> changedModule,
                                             zmf::data::ModuleState newState, zmf::data::ModuleState lastState) {
    // Optional: Handle state changes of modules
}


zmf::data::ZmfOutReply TemplateModule::handleRequest(const zmf::data::ZmfMessage& message,
                                                     const zmf::data::ModuleUniqueId& sender) {
    // Optional: Correctly reply by returning a reply message
    return zmf::data::ZmfOutReply::createNoReply();
}

File: util/template/java/TemplateModule/src/main/java/zsdn/templatemodule/TemplateModule.java
package zsdn.templatemodule;

import java.util.Collection;
import java.util.Collections;

import jmf.data.ModuleDependency;
import jmf.data.ModuleUniqueId;
import jmf.module.AbstractModule;

/**
 * [TODO]
 * 
 * @author [TODO]
 *
 */
public class TemplateModule extends AbstractModule {

	private static final short MODULE_TYPE_ID = 0x0000;
	private static final short VERSION = 0;
	private static final String MODULE_NAME = "TemplateModule";
	private static final Collection<ModuleDependency> DEPENDENCIES = Collections.emptyList();

	public TemplateModule(long instanceId) {
		super(new ModuleUniqueId(MODULE_TYPE_ID, instanceId), VERSION, MODULE_NAME, DEPENDENCIES);
	}

	@Override
	public boolean enable() {
		// TODO enable code
		return true;
	}

	@Override
	public void disable() {
		// TODO disable code
	}
}
